### ISSUE TITLE
Backup Codes

### ISSUE THREAD

**Issue by DeepDiver1975** [owncloud/twofactor_totp#1]
Backup codes are necessary to allow a user to recover from a loss of the second factor device.

**Comment by rmbleeker** [comment]
Seconded. We've implemented OwnCloud at our firm and require users to use 2FA whenever possible. But sometimes users lose or break their phone with the Google Authenticator on it, so backup codes are pretty much a must.

**Comment by karakayasemi** [comment]
I am working on this issue. I added "generate new backup codes" button to setting page. When the user clicks the button, it generates 10 new backup codes.  Each code can be used only once, but the user can generate 10 new backup codes whenever he wants.  But I have a doubt. Should I write this as an external two-factor auth provider app or add "backupCode" provider to this app. For this case, I checked how NextCloud did. They have written it as a core application. Have you any suggestion?
